spack/ci-spack-pyprecice-deps-1804.dockerfile

```diff
@@ -0,0 +1,10 @@
+# Build stage with Spack pre-installed and ready to be used
+FROM spack/ubuntu-bionic:latest
+RUN spack env create precice
+RUN spack env activate precice
+RUN spack install boost@1.74.0 cmake@3.18.4 eigen@3.3.8 petsc@3.14.1
+RUN spack install pkgconfig
+RUN spack env activate precice && spack concretize -f && spack install pkgconfig && spack install precice@develop
+
+# Install python dependencies for pyprecice
+RUN spack install py-setuptools py-wheel py-cython py-numpy py-mpi4py
```

Comment:
Why are you installing pkgconfig in line 6 and 7?

I think the cleanest way of compiling/adding the dependencies to the environment and concretizing them trying to avoid duplication would be:
```
RUN spack env activate precice \ 
    && spack add boost@1.74.0 cmake@3.18.4 eigen@3.3.8 petsc@3.14.1 pkgconfig precice@develop py-setuptools py-wheel py-cython py-numpy py-mpi4py \
    && spack concretize -f \
    && spack install
```

You might specify the environment in a `spack.yaml` file in `${SPACK_ROOT}/var/spack/environments/precice` (where `precice` is the name of the environment). It might help to add the line `concretization: together` in case dependencies are duplicated.

The `spack.yaml` might look like this:
```
# This is a Spack Environment file.
#
# It describes a set of packages to be installed, along with
# configuration settings.
spack:
   specs:
  - boost@1.74.0 
  - cmake@3.18.4 
  - eigen@3.3.8 
  - petsc@3.14.1
  - pkgconfig
  - precice@develop
  - py-setuptools 
  - py-wheel 
  - py-cython 
  - py-numpy
  - py-mpi4py
  view: true
  concretization: together
```

If it works for now, I guess it should be fine. I am not sure if @fsimonis  has an opinion on how to set environments up properly.

Reply:
I changed the dockerfile in 2e22d8c. I will rebuild and test the new file in a moment. But let's keep this threat open until @fsimonis has finished the review. I'm not very experienced with spack and therefore another opinion on the question on "how do I setup an environment?" probably does not harm.

Reply:
Note on `pkgconfig`: Currently preCICE needs `pkgconfig` for finding PETSc. As far as I understand the issue this dependency is currently missing in the package definition of `precice`. Therefore, I'm installing it manually.

Reply:
I guess I missed that this is Docker and every `RUN` is executed in a new container (or so...). So I assume the sole
```
RUN spack install pkgconfig 
```
will install the package in the root environment and 
```
spack env activate precice && spack concretize -f && spack install pkgconfig && spack install precice@develop
```
adds it to the `precice` environment. Regarding the resulting software this should not make any difference. Software that is installed in an environment ends up in the root environment as well.